src/output/dnssim.c

```diff
@@ -216,8 +231,13 @@ void output_dnssim_set_transport(output_dnssim_t* self, output_dnssim_transport_
     case OUTPUT_DNSSIM_TRANSPORT_TCP:
         lnotice("transport set to TCP");
         break;
-    case OUTPUT_DNSSIM_TRANSPORT_UDP:
     case OUTPUT_DNSSIM_TRANSPORT_TLS:
+        lnotice("transport set to TLS");
+#if GNUTLS_VERSION_NUMBER < 0x030603
+        lwarning("TLS session resumption requires GnuTLS 3.6.3+");
```

Comment:
If you don't have TLS resumption, sessions won't be resumed and full handshakes wil be performed instead. If used in benchmarks, this can have serious impact, since full TLS handshakes are way more expensive than resumption.

I wanted to have it on by default and rely on the GnuTLS priority string to disable it. Unfortunately, I don't see any way to determine whether the tickets have been disabled once I set the priority. It's stored in a GnuTLS internal `gnutls_priority_st` and I don't see any API to read the value.

The only other solution I can think of is to introduce a separate variable to control whether session resumption is to be used, independently from GnuTLS priority string.

Reply:
Hmm, tricky situation indeed.

How about check the priority string and if it includes `%NO_TICKETS` then have a warning saying that performance will be heavily affected (regardless of GnuTLS version)?

Reply:
Reason I don't like the warning is because you're assuming it will be used for high performance benchmarking. There are other use-cases where this does not matter and do not warrant a warning.

If you feel that the TLS transport can't be effectively used with older GnuTLS then maybe this should be a compile time check which disabled the entire transport mode.

Reply:
TLS should work just fine with older GnuTLS, I just don't like the inconsistent default behaviour. Perhaps disabling compilation for GnuTLS < 3.6.3 is the way to go - the latest releases of Ubunu, Debian and CentOS all have it and I don't mind if TLS transport would be missing in the older ones.

Reply:
I also ran into some issues when using older libuv. After switching to libuv >= 1.34.2, it was fine. So dnssim might be troublesome to use with older libraries in general.

Reply:
Oh, what kind of inconsistencies?

If it could be explained in the warning (or if the warning point to a section in the man-page that describes the inconsistencies) then maybe it would be OK with the warning.

Reply:
> I also ran into some issues when using older libuv. After switching to libuv >= 1.34.2, it was fine. So dnssim might be troublesome to use with older libraries in general.

That could warrant a warning for the whole module for libuv < 1.34.x.

It might be really good to document these issues so that others don't run into weird things.

Reply:
> Oh, what kind of inconsistencies?

The default tls settings (or any priority string that doesn't contain explicit `%NO_TICKETS`): with GnuTLS >= 3.6.3 it _will_ use session resumption; with older GnuTLS it _won't_ use session resumtpion.

> That could warrant a warning for the whole module for libuv < 1.34.x.
It might be really good to document these issues so that others don't run into weird things.

I think this came up during TCP development, I've switched the version in [this commit](https://gitlab.labs.nic.cz/knot/shotgun/-/commit/593cc1b91ffd6e8efc6f689bf1b7572b197f6a99?merge_request_iid=9). It links to a libuv commit that has the exact assertion error I was getting, but it seems that fix has been in libuv since a long time - I'm not sure if there was a regression or something, but libuv 1.18.0 just didn't work for me.

It'd probably be for the best to add these as compile-time check for dnssim. I could try if it works with 1.24.1 (that's in Debian10).

Reply:
Damn, I can't reproduce the issue with libuv 1.18.0 now. I feel like I'm chasing ghosts! :ghost: Maybe it was caused by some weird bug that used to be in my code?

Anyway, earlier libuv is _probably_ fine. And I'd limit GnuTLS to >= 3.6.3 during compilation to avoid the hassle with older versions.

Reply:
Yeah limit to GnuTLS => 3.6.3 sounds good, are you working on that?

Reply:
Yeah, I'll try it.

Reply:
OK, let me know if you don't have the time and I'll poke at it.

Reply:
I've just pushed my version, feel free to make any adjustments if needed.